amqp/transport.py

```diff
@@ -317,7 +317,7 @@ def _wrap_context(self, sock, sslopts, check_hostname=None, **ctx_options):
 
     def _wrap_socket_sni(self, sock, keyfile=None, certfile=None,
                          server_side=False, cert_reqs=ssl.CERT_NONE,
-                         ca_certs=None, do_handshake_on_connect=True,
+                         ca_certs=None, do_handshake_on_connect=False,
```

Comment:
Does this faulty behaviour occur only in Python 2.7?
If so, we can disable it on Python 2.7 and enable for the rest.

Also, can you please explain what do you mean? Why should we change the default?

Reply:
If this is passed here https://github.com/celery/py-amqp/blob/master/amqp/transport.py#L353 the default python ssl context socket tries to do a handshake before the following code that adds sni headers even executes. The same is true for https://github.com/celery/py-amqp/blob/master/amqp/transport.py#L362 that tries to connect if the flag is set. 

We anyway call this in our code here https://github.com/celery/py-amqp/blob/master/amqp/transport.py#L305 so its better not to pass this as true at this point.